runner/conmon_test/conmon_test.go

```diff
@@ -101,58 +101,170 @@ var _ = Describe("conmon", func() {
 			)
 			Expect(stderr).To(ContainSubstring("Log driver not provided. Use --log-path"))
 		})
+		It("empty log driver should fail", func() {
+			_, stderr := getConmonOutputGivenOptions(
+				conmon.WithPath(conmonPath),
+				conmon.WithContainerID(ctrID),
+				conmon.WithContainerUUID(ctrID),
+				conmon.WithRuntimePath(validPath),
+				conmon.withLogPath(""),
+			)
+			Expect(stderr).To(ContainSubstring("log-path must not be empty"))
+		})
+		It("empty log driver and path should fail", func() {
+			_, stderr := getConmonOutputGivenOptions(
+				conmon.WithPath(conmonPath),
+				conmon.WithContainerID(ctrID),
+				conmon.WithContainerUUID(ctrID),
+				conmon.WithRuntimePath(validPath),
+				conmon.withLogPath(":"),
+			)
+			Expect(stderr).To(ContainSubstring("log-path must not be empty"))
+		})
+		It("k8s-file requires a filename", func() {
+			_, stderr := getConmonOutputGivenOptions(
+				conmon.WithPath(conmonPath),
+				conmon.WithContainerID(ctrID),
+				conmon.WithContainerUUID(ctrID),
+				conmon.WithRuntimePath(validPath),
+				conmon.withLogPath("k8s-file"),
+			)
+			Expect(stderr).To(ContainSubstring("k8s-file requires a filename"))
+		})
+		It("k8s-file: requires a filename", func() {
+			_, stderr := getConmonOutputGivenOptions(
+				conmon.WithPath(conmonPath),
+				conmon.WithContainerID(ctrID),
+				conmon.WithContainerUUID(ctrID),
+				conmon.WithRuntimePath(validPath),
+				conmon.withLogPath("k8s-file:"),
+			)
+			Expect(stderr).To(ContainSubstring("k8s-file requires a filename"))
+		})
 		It("log driver as path should pass", func() {
 			_, stderr := getConmonOutputGivenOptions(
 				conmon.WithPath(conmonPath),
 				conmon.WithContainerID(ctrID),
 				conmon.WithContainerUUID(ctrID),
 				conmon.WithRuntimePath(validPath),
 				conmon.WithLogDriver("", tmpLogPath),
+				conmon.WithLogLevel("debug"),
 			)
-			Expect(stderr).To(BeEmpty())
+			Expect(stderr).To(ContainSubstring("LogDriver K8S TRUE k8s_file "+tmpLogPath))
+			Expect(stderr).To(ContainSubstring("LogDriver JOURNALD FALSE"))
```

Comment:
I'm not super convinced I like checking against this. I would rather check against what's actually in the file/journal like we do [here](https://github.com/containers/conmon/pull/148)

Reply:
if you want, we can just drop checks to this now, and later implement a more complete check that the journal/file is actually being written to

Reply:
My intention here was that the debug logging indicates that it will actually do what I requested it to do.  (Hence, given the arguments I provided and the expected behavior, conmon's configured state matches that)

I did it on multiple log lines so additional log drivers could be added without breaking the check.

Checking that the file or journal is actually written to - well, I'm not against that, I just only ever saw empty files from the unit test and wasn't sure how to get conmon to log something :-D i.e. Mocking a "runc" that would instead echo something specific.

Reply:
Dun